### PR TITLE
Add variables to Flash Patcher

### LINKED ISSUE #78: Add variables
Add scoped variables (with `set VARNAME=value`) and global variables (with `export VARNAME=value`). When typing `${VARNAME}` we should do a simple find and replace with the variable value.

The scope should be any file, or anything called within a specific file.

### PR DESCRIPTION
**Pull request by rayyaw**
## Changes
- Add support for defining and using variables.

## Testing
- Automated testing.

## Related issues
- closes https://github.com/rayyaw/flash-patcher/issues/78

### CODE REVIEW

File: flash_patcher/parse/scope.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        """Resolve all instances of variables within a string,
+        and return it with the content replaced.
+        """
+        variable_use = re.compile(r"\${.*?}")
+
+        matches = variable_use.findall(content)
+
+        resolved_content = content
+
+        # we can't use match as a variable name since it's a keyword
```
Comment (GTcreyon) [inline flash_patcher/parse/scope.py]: Prefer capitals to begin comments.

File: test/parse/test_scope.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+from unittest import TestCase
+
+from pytest import raises
+
+from flash_patcher.exception.error_manager import ErrorManager
+from flash_patcher.parse.scope import Scope
+
+class ScopeSpec (TestCase):
+
```
Comment (GTcreyon) [inline test/parse/test_scope.py:11]: Unnecessary whitespace?

**Review by GTcreyon — APPROVED** [review]

File: test/parse/test_scope.py (hunk 2 above)
Comment (rayyaw) [reply test/parse/test_scope.py:11]: This seems to be standard across our test cases, will keep as is

File: flash_patcher/parse/scope.py (hunk 1 above)
Comment (rayyaw) [reply flash_patcher/parse/scope.py]: noted, will fix
